src/pages/Fox/components/FoxFarming.tsx

```diff
@@ -0,0 +1,365 @@
+import {
+  selectAssetById,
+  selectEarnUserStakingOpportunityByOpportunityId,
+  selectMarketDataByAssetIdUserCurrency,
+} from 'state/slices/selectors'
+import { useAppDispatch, useAppSelector } from 'state/store'
+
+import { useFoxPageContext } from '../hooks/useFoxPageContext'
+
+const containerPaddingX = { base: 4, xl: 0 }
+const headerTitleMb = { base: 4, md: 0 }
+const headerTitleMaxWidth = { base: '100%', md: '50%' }
+const headerSx: FlexProps['sx'] = {
+  alignItems: { base: 'flex-start', md: 'center' },
+  justifyContent: 'space-between',
+  mb: 8,
+  flexDir: {
+    base: 'column',
+    md: 'row',
+  },
+}
+const stackProps: StackProps = {
+  width: '100%',
+  flexDir: 'column',
+  flex: 1,
+  spacing: 0,
+}
+const columnsProps = {
+  base: 1,
+  md: 2,
+}
+
+export const FoxFarming = () => {
+  const { assetAccountId } = useFoxPageContext()
+  const translate = useTranslate()
+  const isFoxFarmingEnabled = useFeatureFlag('FoxPageFoxFarmingSection')
+  const history = useHistory()
+  const location = useLocation()
+  const appDispatch = useAppDispatch()
+  const {
+    state: { isConnected, isDemoWallet },
+    dispatch,
+  } = useWallet()
+  const queryClient = useQueryClient()
+
+  useEffect(() => {
+    appDispatch(marketApi.endpoints.findByAssetId.initiate(foxEthLpAssetId))
+  }, [appDispatch])
+
+  const foxEthMarketData = useAppSelector(state =>
+    selectMarketDataByAssetIdUserCurrency(state, foxEthLpAssetId),
+  )
+
+  const opportunityId = useMemo(
+    () =>
+      toOpportunityId({
+        assetNamespace: 'erc20',
+        assetReference: ETH_FOX_STAKING_EVERGREEN_CONTRACT,
+        chainId: assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId,
+      }),
+    [assetAccountId],
+  )
+
+  const stakingOpportunityByIdFilter = useMemo(() => {
+    const userStakingId = assetAccountId
+      ? serializeUserStakingId(assetAccountId, opportunityId)
+      : undefined
+    return { userStakingId, opportunityId }
+  }, [assetAccountId, opportunityId])
+
+  const opportunity = useAppSelector(state =>
+    selectEarnUserStakingOpportunityByOpportunityId(state, stakingOpportunityByIdFilter),
+  )
+
+  const { getPeriodFinish } = useFoxFarming(ETH_FOX_STAKING_EVERGREEN_CONTRACT)
+
+  const {
+    data: periodFinish,
+    isLoading: isGetPeriodFinishQueryLoading,
+    isFetching: isGetPeriodFetchingLoading,
+  } = useQuery({
+    queryKey: ['getPeriodFinish'],
+    queryFn: () => getPeriodFinish(),
+  })
+
+  const isGetPeriodFinishLoading = useMemo(
+    () => isGetPeriodFinishQueryLoading || isGetPeriodFetchingLoading,
+    [isGetPeriodFinishQueryLoading, isGetPeriodFetchingLoading],
+  )
+
+  const fetchOpportunityData = useCallback(async () => {
+    if (!foxEthMarketData) return
+
+    const chainId = assetAccountId ? fromAccountId(assetAccountId).chainId : ethChainId
+
+    await fetchAllOpportunitiesIdsByChainId(appDispatch, chainId)
```

Comment:
blocking: this thunk will dispatch fetch of OpportunityIds for all opportunities this ChainId supports, we should be able to do this instead:
```suggestion
   await dispatch(
    getOpportunityIds.initiate(
      {
        defiProvider: DefiProvider.EthFoxStaking,
        defiType: DefiType.Staking,
      },
      options,
    ),
  )
```

Reply:
Done in [527f86c](https://github.com/shapeshift/web/pull/7934/commits/527f86ce55071bc81f46808931a053d3e5ee04e3)